### ISSUE TITLE
the original input cannot be reconstructed from the tokenized output: token.whitespace is empty when the space char is expected

### ISSUE THREAD

**Issue by vitaly-d** [explosion/spaCy#6663]
## How to reproduce the behaviour
Hi! I  [converted](https://github.com/explosion/spaCy/discussions/6435#discussioncomment-186017) our in-house jsonl NER corpus and noticed a bunch of W030. I've [read](https://github.com/explosion/spaCy/discussions/6240#discussioncomment-189592) that this warning could be caused by a misaligned entity, but this case is different: the original jsonl data seems to be correct and misaligned entities appeared due a tokenisation issue: in some rare cases the original input cannot be reconstructed from the tokenized output:

```
import spacy
nlp = spacy.blank("en")
text="A): B): C"
assert text == nlp(text).text, f"{text} != {nlp(text)}"

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-d41e06da1b07> in <module>
      2 nlp = spacy.blank("en")
      3 text="A): B): C"
----> 4 assert text == nlp(text).text, f"{text} != {nlp(text)}"

AssertionError: A): B): C != A): B):C
```

```
In [98]: [ (t.text, t.whitespace_) for t in nlp(text)]
Out[98]: [('A', ''), ('):', ' '), ('B', ''), ('):', ''), ('C', '')]
```

## Info about spaCy
- **spaCy version:** 3.0.0rc2
- **Platform:** Linux-5.4.0-54-generic-x86_64-with-glibc2.10
- **Python version:** 3.8.5
- **Pipelines:** en_core_web_trf (3.0.0a0), en_core_web_lg (3.0.0a0)

_This issue can be reproduced only with Spacy v3._

**Comment by adrianeboyd** [comment]
Yes, this is a bug. I think it should be fixed by https://github.com/explosion/spaCy/pull/6475, if you'd like to try out building v3 from `develop` from source. Otherwise the fix should be in the next release, v3.0.0rc3.

**Comment by vitaly-d** [comment]
Thank you! v3 from develop is free of the bug.
